test/meson.build

```diff
@@ -117,8 +117,16 @@ test('paths', paths)
 if get_option('libmpv')
     exe = executable('libmpv-test', 'libmpv_test.c',
                      include_directories: incdir, link_with: libmpv)
+
+    # Generate files for testing if possible. ffmpeg 4 generates incorrect output so skip.
+    ffmpeg = find_program('ffmpeg', required: false)
+    if ffmpeg.found() and ffmpeg.version().version_compare('>=0.5')
+        generate_test_media = find_program(join_paths(source_root, 'ci', 'generate-test-media.sh'), required: true)
+        run_command(generate_test_media, build_root, check: true)
```

Comment:
I would use `custom_target` to generate it on demand and not on every configure.

Reply:
Hmm, `custom_target` is really annoying in this context.

Reply:
I have not tried myself. But wouldn't it be 

```
test_files = []
test_file += custom_target('eng_default.mkv',
    output: 'eng_default.mkv',
    command: [ffmpeg, ..., '@OUTPUT@'],
)
```
and then 
```
test('libmpv', exe, args: [file, test_files], timeout: 60, should_fail: not features['lua'])
```
remove 
```
const char *files[]
```
array and ready input files from argv.

Reply:
Generating files on every configure unconditionally with external ffmpeg is no-go for me. Exact solution is up in the air, but I would prefer it not do complex stuff in configure.

Reply:
It would work but you would need a different target for every single generated file which feels kind of stupid. But I haven't thought of anything else better yet.

Reply:
Well, for me it is natural, they are different commads after all